### PR TITLE
attach: add LXC_ATTACH_ALLOCATE_PTY

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: doc/lxc-attach.sgml.in [hunk 1]
```diff
@@ -58,7 +58,6 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
       <arg choice="opt">-R, --remount-sys-proc</arg>
       <arg choice="opt">--keep-env</arg>
       <arg choice="opt">--clear-env</arg>
-      <arg choice="opt">-L, --pty-log <replaceable>file</replaceable></arg>
```
Comment (hallyn) [inline doc/lxc-attach.sgml.in]: This seems like a backward-incompatible change in behavior.  Any sites using lxc-attach -L in a script will end up with breakage?

**Comment by brauner** [comment]
On Jan 5, 2018 04:07, "Serge Hallyn" <notifications@github.com> wrote:

*@hallyn* commented on this pull request.
------------------------------

In doc/lxc-attach.sgml.in
<https://github.com/lxc/lxc/pull/2061#discussion_r159803350>:

> @@ -58,7 +58,6 @@ Foundation, Inc., 51 Franklin Street <https://maps.google.com/?q=51+Franklin+Street&entry=gmail&source=g>, Fifth Floor, Boston, MA 02110-1301 USA
       <arg choice="opt">-R, --remount-sys-proc</arg>
       <arg choice="opt">--keep-env</arg>
       <arg choice="opt">--clear-env</arg>
-      <arg choice="opt">-L, --pty-log <replaceable>file</replaceable></arg>

This seems like a backward-incompatible change in behavior. Any sites using
lxc-attach -L in a script will end up with breakage?


There's no clear way to make this work without calling non-api symbols.
Since this is 3.0 small breakages are fine I think. We can however, if this
feels better to you retain the flag but make it a noop.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/lxc/lxc/pull/2061#pullrequestreview-86814913>, or mute
the thread

**Comment by hallyn** [comment]
Sorry, you say "this is 3.0" - I'm confused, this is being applied to lxc/master so won't wait for 3.0 right?

**Comment by brauner** [comment]
There'll be no feature release before `3.0` anymore. At least it isn't planned.

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
This needs to be merged soon so that I can go on splitting the tools.

File: doc/lxc-attach.sgml.in (hunk 1 above)
Comment (brauner) [reply doc/lxc-attach.sgml.in]: An argument has been added to `lxc_attach_options_t`.

**Pull request merged**